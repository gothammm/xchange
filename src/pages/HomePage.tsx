import React, { useState } from 'react';
import FlexContainer from '../components/FlexContainer';
import styled from 'styled-components';
import { PlusOutlined, SyncOutlined } from '@ant-design/icons';
import XChangeButton from '../components/XChangeButton';
import { useWallet } from '../hooks/WalletHook';
import WalletOverview from '../components/WalletOverview';
import { Drawer, message } from 'antd';
import WalletList from '../components/WalletList';
import { DrawerProps } from 'antd/lib/drawer';
import AddMoneyModal from '../components/AddMoneyModal';

const HomeContainer = styled(FlexContainer)`
  justify-content: center;
  align-items: center;
  flex: 0.9;
`;

const XChangeDrawer: React.FC<DrawerProps> = styled(Drawer)`
  .ant-drawer-body {
    padding: 0 8px;
  }
`;

const WalletContainer = styled.div`
  margin: 3em 0;
`;

const ActionBarContainer = styled(FlexContainer)`
  justify-content: space-between;
  flex: none;
  flex-direction: row;
  div {
    margin: 0 2.5em;
  }
`;

const HomePage: React.FC = () => {
  const { wallets, resetWallet, makePrimary, addMoneyToWallet } = useWallet();
  const primaryWallet = wallets?.find(wallet => wallet.isPrimary);
  const [showWallets, setShowWallets] = useState(false);
  const [showAddMoney, setShowAddMoney] = useState(false);
  return (
    <HomeContainer>
      {primaryWallet && (
        <AddMoneyModal
          show={showAddMoney}
          wallet={primaryWallet}
          onCancel={() => setShowAddMoney(false)}
          onMoneyAdd={(value: number) => {
            addMoneyToWallet(primaryWallet, value);
            setShowAddMoney(false);
            message.success(`Money added successfully!`);
          }}
        />
      )}
      <WalletContainer>
        <WalletOverview
          onWalletCreateClick={() => {
            resetWallet();
            message.success(`Wallet reset to initial state.`);
          }}
          wallet={primaryWallet}
          onWalletClick={() => setShowWallets(true)}
        />
      </WalletContainer>
      <ActionBarContainer>
        <XChangeButton
          type="primary"
          shape="circle"
          icon={<PlusOutlined />}
          size={'large'}
          text={'Add Money'}
          onClick={() => setShowAddMoney(true)}
        />
        <XChangeButton
          type="primary"
          shape="circle"
          icon={<SyncOutlined />}
          text={'Exchange'}
          size={'large'}
        />
      </ActionBarContainer>
      <XChangeDrawer
        title="Available Wallets"
        height={300}
        placement="bottom"
        closable={false}
        onClose={() => setShowWallets(false)}
        visible={showWallets}
      >
        <WalletList
          wallets={wallets || []}
          onMakeWalletPrimary={wallet => {
            makePrimary(wallet);
            setShowWallets(false);
          }}
        />
      </XChangeDrawer>
    </HomeContainer>
  );
};

export default HomePage;
