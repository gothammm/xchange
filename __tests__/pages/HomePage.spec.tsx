import React from 'react';
import { render } from '@testing-library/react';
import HomePage from '../../src/pages/HomePage';

describe(`CurrencyDisplay component`, () => {
  test(`should match snapshot for various currency type`, () => {
    const component = render(<HomePage />);
    expect(component.asFragment()).toMatchSnapshot();
  });
});
