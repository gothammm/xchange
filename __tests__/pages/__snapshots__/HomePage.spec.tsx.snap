// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm sc-fzpans fKdfki"
  >
    <div
      class="sc-fznyAO kgVGNT"
    >
      <div
        class="sc-AxjAm sc-AxhCb dgqfQu"
      >
        <div
          class="ant-empty"
        >
          <div
            class="ant-empty-image"
          >
            <svg
              class="ant-empty-img-default"
              height="152"
              viewBox="0 0 184 152"
              width="184"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fill-rule="evenodd"
              >
                <g
                  transform="translate(24 31.67)"
                >
                  <ellipse
                    class="ant-empty-img-default-ellipse"
                    cx="67.797"
                    cy="106.89"
                    rx="67.797"
                    ry="12.668"
                  />
                  <path
                    class="ant-empty-img-default-path-1"
                    d="M122.034 69.674L98.109 40.229c-1.148-1.386-2.826-2.225-4.593-2.225h-51.44c-1.766 0-3.444.839-4.592 2.225L13.56 69.674v15.383h108.475V69.674z"
                  />
                  <path
                    class="ant-empty-img-default-path-2"
                    d="M101.537 86.214L80.63 61.102c-1.001-1.207-2.507-1.867-4.048-1.867H31.724c-1.54 0-3.047.66-4.048 1.867L6.769 86.214v13.792h94.768V86.214z"
                    transform="translate(13.56)"
                  />
                  <path
                    class="ant-empty-img-default-path-3"
                    d="M33.83 0h67.933a4 4 0 0 1 4 4v93.344a4 4 0 0 1-4 4H33.83a4 4 0 0 1-4-4V4a4 4 0 0 1 4-4z"
                  />
                  <path
                    class="ant-empty-img-default-path-4"
                    d="M42.678 9.953h50.237a2 2 0 0 1 2 2V36.91a2 2 0 0 1-2 2H42.678a2 2 0 0 1-2-2V11.953a2 2 0 0 1 2-2zM42.94 49.767h49.713a2.262 2.262 0 1 1 0 4.524H42.94a2.262 2.262 0 0 1 0-4.524zM42.94 61.53h49.713a2.262 2.262 0 1 1 0 4.525H42.94a2.262 2.262 0 0 1 0-4.525zM121.813 105.032c-.775 3.071-3.497 5.36-6.735 5.36H20.515c-3.238 0-5.96-2.29-6.734-5.36a7.309 7.309 0 0 1-.222-1.79V69.675h26.318c2.907 0 5.25 2.448 5.25 5.42v.04c0 2.971 2.37 5.37 5.277 5.37h34.785c2.907 0 5.277-2.421 5.277-5.393V75.1c0-2.972 2.343-5.426 5.25-5.426h26.318v33.569c0 .617-.077 1.216-.221 1.789z"
                  />
                </g>
                <path
                  class="ant-empty-img-default-path-5"
                  d="M149.121 33.292l-6.83 2.65a1 1 0 0 1-1.317-1.23l1.937-6.207c-2.589-2.944-4.109-6.534-4.109-10.408C138.802 8.102 148.92 0 161.402 0 173.881 0 184 8.102 184 18.097c0 9.995-10.118 18.097-22.599 18.097-4.528 0-8.744-1.066-12.28-2.902z"
                />
                <g
                  class="ant-empty-img-default-g"
                  transform="translate(149.65 15.383)"
                >
                  <ellipse
                    cx="20.654"
                    cy="3.167"
                    rx="2.849"
                    ry="2.815"
                  />
                  <path
                    d="M5.698 5.63H0L2.898.704zM9.259.704h4.985V5.63H9.259z"
                  />
                </g>
              </g>
            </svg>
          </div>
          <p
            class="ant-empty-description"
          >
            No wallet or any primary wallet found.
          </p>
          <div
            class="ant-empty-footer"
          >
            <button
              class="ant-btn ant-btn-primary"
              type="button"
            >
              <span>
                Create wallet now
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-AxjAm sc-fznKkj cXnrRU"
    >
      <div
        class="sc-AxjAm sc-AxirZ gtbJzn"
      >
        <button
          class="ant-btn ant-btn-primary ant-btn-circle ant-btn-lg ant-btn-icon-only"
          text="Add Money"
          type="button"
        >
          <span
            aria-label="plus"
            class="anticon anticon-plus"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="plus"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <defs>
                <style />
              </defs>
              <path
                d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
              />
              <path
                d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
              />
            </svg>
          </span>
        </button>
        <span
          class="sc-AxiKw hKYpuK"
        >
          Add Money
        </span>
      </div>
      <div
        class="sc-AxjAm sc-AxirZ gtbJzn"
      >
        <button
          class="ant-btn ant-btn-primary ant-btn-circle ant-btn-lg ant-btn-icon-only"
          text="Exchange"
          type="button"
        >
          <span
            aria-label="sync"
            class="anticon anticon-sync"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="sync"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M168 504.2c1-43.7 10-86.1 26.9-126 17.3-41 42.1-77.7 73.7-109.4S337 212.3 378 195c42.4-17.9 87.4-27 133.9-27s91.5 9.1 133.8 27A341.5 341.5 0 01755 268.8c9.9 9.9 19.2 20.4 27.8 31.4l-60.2 47a8 8 0 003 14.1l175.7 43c5 1.2 9.9-2.6 9.9-7.7l.8-180.9c0-6.7-7.7-10.5-12.9-6.3l-56.4 44.1C765.8 155.1 646.2 92 511.8 92 282.7 92 96.3 275.6 92 503.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8zm756 7.8h-60c-4.4 0-7.9 3.5-8 7.8-1 43.7-10 86.1-26.9 126-17.3 41-42.1 77.8-73.7 109.4A342.45 342.45 0 01512.1 856a342.24 342.24 0 01-243.2-100.8c-9.9-9.9-19.2-20.4-27.8-31.4l60.2-47a8 8 0 00-3-14.1l-175.7-43c-5-1.2-9.9 2.6-9.9 7.7l-.7 181c0 6.7 7.7 10.5 12.9 6.3l56.4-44.1C258.2 868.9 377.8 932 512.2 932c229.2 0 415.5-183.7 419.8-411.8a8 8 0 00-8-8.2z"
              />
            </svg>
          </span>
        </button>
        <span
          class="sc-AxiKw hKYpuK"
        >
          Exchange
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
